theano/gpuarray/dnn.py

```diff
@@ -530,6 +531,7 @@ def __init__(self, algo=None, inplace=False):
         self.choose_algo = self.algo in SUPPORTED_DNN_CONV_ALGO_RUNTIME
         self.choose_once = self.algo in DNN_CONV_ALGO_CHOOSE_ONCE
         self.choose_time = self.algo in DNN_CONV_ALGO_CHOOSE_TIME
+        self.num_groups = num_groups
 
     def __setstate__(self, d):
```

Comment:
Just a suggestion: I think you should also update `__setstate__()`, as you add a new class attribute (`num_groups`) that may not exist in old pickled instances of this Op. That may causes some errors, e.g. in FAST_COMPILE mode.

Same for every other Op where you add a new attribute.